### DISCUSSION TITLE
How to set Copilot Edits save files when Accept all changes?

### DISCUSSION THREAD

**Discussion by doggy8088** [community/community#148084] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have to save each files after Accept all Copilot Edits changes. Is it possible to save all files when I accept all changes automatically?

**Comment by NadaAbuIssa** [comment]
Yes it is, go to the settings search for alwaysSaveWithGeneratedChanges and enable it

**Reply by doggy8088** [reply]
I tried `chat.editing.alwaysSaveWithGeneratedChanges` setting. It's not working as expected in Copilot Edits..

**Comment by doggy8088** [comment, marked as answer]
I finally figured out. Here is my explanation.

If you have used Copilot Edits to generate code, when you press **Accept**, it does not automatically save. However, there is a **Save All** button that, when pressed, will save files including AI-generated changes, but a confirmation dialog will appear before saving, helping you to save everything. After saving, you can press **Accept** to complete the edit. You can also press **Discard** to revert changes, even if you have already saved, you can still revert to the pre-edit state through the **Discard** button.

The above behavior is default because many projects might rely on automated tasks triggered by file changes, such as automatic building or testing tasks, like `dotnet watch`, `ng serve --watch`, `webpack`, or `parcel watch`, which have similar mechanisms. If it were to autosave, these tools might be triggered continuously.

In the dialog box of hitting the **Save All** button, if you check the option **Always save with AI-generated changes without asking**, you can save files without asking for confirmation in the future. After checking, the original **Save All** button will be hidden. In fact, this action adjusts the `chat.editing.alwaysSaveWithGeneratedChanges` setting to `true`.

After setting `chat.editing.alwaysSaveWithGeneratedChanges` to `true`, the next time you edit a file with Copilot Edits, it still won't **Autosave** before or after you press **Accept** button. Unless you enable VS Code's **[File]** > **[Auto Save]** feature, it will truly **Autosave**!
